static/third/bootstrap-typeahead/typeahead.js

```diff
@@ -348,6 +357,7 @@
 
         case 27: // escape
           if (!this.shown) return
+          $(document).trigger("zuliptypeaheadclosed"); // Zulip specific change.
```

Comment:
We are generally trying to phase out calls to `$(document).trigger` in our codebase, since they lead to kind of hard-to-test and hard-to-find anonymous functions.  I realize this is sort of a special case, since we're dealing with a third party library, and having it call Zulip code directly would be pretty ugly.  But is there no reason we can't just have some kind of callback that we register here when you initialize the widget?

Reply:
We can add an option like "onDismissedCallback" which would be a no-op function unless provided when setting up the typeahead. It would be a slightly more intrusive change to the library which is why I first tried this approach.